rules/prefer-string-slice.js

```diff
@@ -10,6 +10,24 @@ const argumentsVariable = templates.spreadVariable();
 const substrCallTemplate = templates.template`${objectVariable}.substr(${argumentsVariable})`;
 const substringCallTemplate = templates.template`${objectVariable}.substring(${argumentsVariable})`;
 
+const getNumericValue = node => {
+	if (node.type === 'Literal' && typeof node.value === 'number') {
+		return node.value;
+	}
+
+	if (node.type === 'UnaryExpression' && node.operator === '-') {
+		return 0 - getNumericValue(node.argument);
+	}
+};
+
+const isLengthProperty = node => (
```

Comment:
I don't think we  want this case, just  number  and  other

Reply:
I found this useful when applying this on a large project of mine. Can you explain why you don't want it? Because you think it will likely result in invalid fixes? Because you think the use case is too specific to merit the complexity of it? Something else?

Reply:
yes, it's useful in some case,  because it's
 always  >=0, so no need compare with 0, but in js word maybe too many cases to handle , like `Set#size()` , `Math.abs` also works, we don't want to handle all of them. so why .length special?

I wounder what  @sindresorhus think

Reply:
`.length` is what's used on strings and these methods are oriented on strings, so not that unlikely that they reference the length of a string. Eg. writing `.slice('foo'.length)` can be more easy to understand than writing `.slice(3)` and similarly eg:

```javascript
if (bar.startsWith(SOME_PREFIX)) {
  bar = bar.slice(SOME_PREFIX.length);
}
```

Reply:
I think it covers a common use-case and that we should keep this.